lib/ecto/uuid.ex

```diff
@@ -183,24 +204,49 @@ defmodule Ecto.UUID do
     end
   end
 
+  @default_version 4
+  @default_options version: 4
   @doc """
-  Generates a random, version 4 UUID.
+  Generates a uuid with the given options.
   """
   @spec generate() :: t
-  def generate(), do: encode(bingenerate())
+  @spec generate(options) :: t
+  def generate(opts \\ [@default_options]), do: encode(bingenerate(opts))
 
   @doc """
-  Generates a random, version 4 UUID in the binary format.
+  Generates a version 4 uuid in binary format.
   """
   @spec bingenerate() :: raw
-  def bingenerate() do
+  def bingenerate(), do: bingenerate(@default_options)
+
+  @doc """
+  Generates a uuid with the given options in binary format.
+  """
+  @spec bingenerate(options) :: raw
+  def bingenerate(opts) do
+    case Keyword.get(opts, :version, @default_version) do
+      4 -> bingenerate_v4()
+      7 -> bingenerate_v7()
+      _ -> raise ArgumentError, "unknown UUID version: #{inspect(opts[:version])}"
+    end
+  end
```

Comment:
same idea as generate

```suggestion
  @doc """
  Generates a uuid with the given options in binary format.
  """
  @spec bingenerate(options) :: raw
  def bingenerate(opts \\ []) do
    case Keyword.get(opts, :version, @default_version) do
      4 -> bingenerate_v4()
      7 -> bingenerate_v7()
      version -> raise ArgumentError, "unknown UUID version: #{inspect(version)}"
    end
  end
```